src/shims/x86/sse2.rs

```diff
@@ -0,0 +1,1042 @@
+
+                let (left, left_len) = this.operand_to_simd(left)?;
+                let (right, _) = this.operand_to_simd(right)?;
+                let (dest, dest_len) = this.place_to_simd(dest)?;
+
+                assert_eq!(dest_len, left_len);
+
+                // Convert first element of `right`
+                let res0 = this
+                    .read_scalar(&this.project_index(&right, 0)?)?
+                    .to_f64()?
+                    .convert(/*loses_info*/ &mut false)
+                    .value;
+                this.write_scalar(Scalar::from_f32(res0), &this.project_index(&dest, 0)?)?;
+
+                // Copy remianing from `left`
+                for i in 1..dest_len {
+                    this.copy_op(
+                        &this.project_index(&left, i)?,
+                        &this.project_index(&dest, i)?,
+                        /*allow_transmute*/ false,
+                    )?;
+                }
+            }
+            // Used to implement the _mm_cvtss_sd function.
+            // Converts the first f64 from `right` to f32 and copies
+            // the remaining elements from `left`
+            "cvtss2sd" => {
+                let [left, right] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+
+                let (left, left_len) = this.operand_to_simd(left)?;
+                let (right, _) = this.operand_to_simd(right)?;
+                let (dest, dest_len) = this.place_to_simd(dest)?;
+
+                assert_eq!(dest_len, left_len);
+
+                // Convert first element of `right`
+                let res0 = this
+                    .read_scalar(&this.project_index(&right, 0)?)?
+                    .to_f32()?
+                    .convert(/*loses_info*/ &mut false)
+                    .value;
+                this.write_scalar(Scalar::from_f64(res0), &this.project_index(&dest, 0)?)?;
+
+                // Copy remianing from `left`
+                for i in 1..dest_len {
+                    this.copy_op(
+                        &this.project_index(&left, i)?,
+                        &this.project_index(&dest, i)?,
+                        /*allow_transmute*/ false,
+                    )?;
+                }
+            }
+            // Used to implement the _mm_movemask_pd function.
+            // Returns a scalar integer where the i-th bit is the highest
+            // bit of the i-th component of `op`.
+            // https://www.felixcloutier.com/x86/movmskpd
+            "movmsk.pd" => {
+                let [op] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let (op, op_len) = this.operand_to_simd(op)?;
+
+                let mut res = 0;
+                for i in 0..op_len {
+                    let op = this.read_scalar(&this.project_index(&op, i)?)?;
+                    let op = op.to_u64()?;
+
+                    res |= (op >> 63) << i;
```

Comment:
Isn't there a nicer way to say "highest bit"? At least please add a comment saying that this is the highest bit.